File: 2/2_1.go
package main

import (
	"os"
	"fmt"
	"bufio"
	"strconv"
	"strings"
)

func main() {
	var horiz int = 0
	var depth int = 0

	file, err := os.Open("data.txt")
	if err != nil {
		fmt.Println(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		fields := strings.Fields(scanner.Text())
		direction := fields[0]
		amt, _ := strconv.Atoi(fields[1])
		if direction ==  "forward" {
			horiz += amt
			continue
		}

		if direction == "up" {
			amt *= -1
		}

		depth += amt
	}

	fmt.Println(horiz * depth)
}

File: 2/2_1.py
#!/usr/bin/python3
from pprint import pprint

horiz = 0
depth = 0

with open('data.txt') as file:
    for line in file:
        direction, amt = line.rstrip().split(" ")
        amt = int(amt)
        if direction == 'forward':
            horiz += amt
        else:
            amt = amt if direction == 'down' else amt * -1
            depth += amt

pprint(horiz * depth)
